### DISCUSSION TITLE
The image is not full width in large viewports

### DISCUSSION THREAD

**Discussion by parsagholipour** [igordanchenko/yet-another-react-lightbox#182] [General]
When I inspect in Chrome and resize the viewport to a big value like width = 1700px the image becomes small and resizing back doesn't fix it too.
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/d8218016-b6a1-4b5f-aafc-ceaa89e07ac3)
You can check it here
https://arcane-fortress.com/products/stained-glass-radiant-100-deck-box/
It gets fixed by applying the following styles to the img element:
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/56330e3d-cb61-49be-a1c7-ec3ffb87ccf2)

**Comment by igordanchenko** [comment]
It looks like you are not providing `height` attributes in the `srcSet` array elements. 

<img width="615" alt="Screenshot 2023-09-02 at 7 06 30 PM" src="https://github.com/igordanchenko/yet-another-react-lightbox/assets/64441155/363cf2a5-b5f8-4d27-a822-39ac5968164e">

Adding `height` attributes should address this issue.

**Reply by parsagholipour** [reply]
It didn't work. I logged the Yarl slides prop. The height is provided now.

**Reply by igordanchenko** [reply]
Can you post an isolated test case on Codesandbox? The above page no longer works...

**Reply by parsagholipour** [reply]
Can you check the link again please

**Reply by igordanchenko** [reply]
Hmmm... I'm still unable to reproduce this issue locally. Interesting observation - this happens only in Chrome (Safari and Firefox render as expected)

Can you add trace logging with  the following [patch-package](https://www.npmjs.com/package/patch-package) patch - https://gist.github.com/igordanchenko/63cabe372a3160efba1dd9a74c6b1ee0

**Reply by parsagholipour** [reply]
I installed the patch.
Some logs when this problem occurred:
[chrome.log](https://github.com/igordanchenko/yet-another-react-lightbox/files/12506655/chrome.log)
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/ceb28b40-baaa-4b5f-9d03-9439e408b3c3)
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/0b8eb44f-1410-4fc0-bb1b-9df6a936f6df)

The image is smaller than the container:
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/eebc4037-1e6e-43c1-a607-89ae301c99ea)
and giving `width: 100%` fixes.

**Reply by igordanchenko** [reply]
Thank you for adding trace logging.

Normally, the lightbox controller renders with `containerRect: undefined` only one time during the initial render. However, in your case this happens 3 times as if the lightbox re-mounts and renders again from scratch. 

<img width="355" alt="Screenshot 2023-09-03 at 3 49 26 PM" src="https://github.com/igordanchenko/yet-another-react-lightbox/assets/64441155/1b328a7e-13d9-4bde-befd-fd61298e3142">

However, I'm not sure it actually contributes to the issue you are facing.

Anyway, it looks like the following CSS style should do the trick in your case:

```css
.custom_yarl .yarl__slide_image {
  flex: 1;
}
```

**Reply by parsagholipour** [reply]
Yes, because I added  `key={windowWidth}` hoping it fixes the issue.
The css alone doesn't fix. Should also unset max height.
![image](https://github.com/igordanchenko/yet-another-react-lightbox/assets/57809920/a6413e43-7757-41cc-a134-3a4e26f1fa71)

**Reply by igordanchenko** [reply]
> The css alone doesn't fix. Should also unset max height.

Not really. You are not passing `height` prop in the `srcSet` again. You won't need to unset the `maxWidth` and `maxHeight` once you add the `height` attribute back.

The lightbox incorrectly calculates the `maxHeight` in this case. I'll add a safeguard against the invalid `srcSet` data.
